corehq/apps/users/models.py

```diff
@@ -1179,7 +1175,7 @@ def delete(self, deleted_by, deleted_via=None):
             user = self.get_django_user()
             user.delete()
             if deleted_by:
-                log_model_change(user, deleted_by, message=f"deleted_via: {deleted_via}",
+                log_model_change(deleted_by, user, message=f"deleted_via: {deleted_via}",
```

Comment:
@mkangia is this tested anywhere? This was giving me errors because the args were the wrong way around.

Reply:
damn! that's bad.

logging on delete is not tested. I have added a test in the new PR i am working on and also fixed it there, unknowingly.
will follow up if there are entries on prod that have this other way round and see how it can be fixed.

Reply:
was investigating more. I see now, you meant this was never running,
I see the error locally
```
  File "/home/mkangia/Work/dimagi/commcare-hq/corehq/util/model_log.py", line 40, in log_model_change
    content_type_id=get_content_type_for_model(model_object).pk,
  File "/home/mkangia/venv/cchq/lib/python3.6/site-packages/django/contrib/admin/options.py", line 66, in get_content_type_for_model
    return ContentType.objects.get_for_model(obj, for_concrete_model=False)
  File "/home/mkangia/venv/cchq/lib/python3.6/site-packages/django/contrib/contenttypes/models.py", line 42, in get_for_model
    return self._get_from_cache(opts)
  File "/home/mkangia/venv/cchq/lib/python3.6/site-packages/django/contrib/contenttypes/models.py", line 31, in _get_from_cache
    key = (opts.app_label, opts.model_name)
AttributeError: 'Options' object has no attribute 'model_name'
```

Is it safe to assume that it was never called?
https://sentry.io/organizations/dimagi/issues/?query=is%3Aunresolved+has+no+attribute+%27model_name%27&statsPeriod=14d

Reply:
I was getting as different error:
```
File "/home/skelly/src/cchq/corehq/apps/api/resources/v0_5.py", line 383, in obj_create
    bundle.obj.delete(deleted_by=request_user, deleted_via=USER_CHANGE_VIA_API)
  File "/home/skelly/src/cchq/corehq/apps/users/models.py", line 1178, in delete
    log_model_change(user, deleted_by, message=f"deleted_via: {deleted_via}",
  File "/home/skelly/src/cchq/corehq/util/model_log.py", line 37, in log_model_change
    return LogEntry.objects.log_action(
  File "/home/skelly/.virtualenvs/hq/lib/python3.8/site-packages/django/contrib/admin/models.py", line 29, in log_action
    return self.model.objects.create(
  File "/home/skelly/.virtualenvs/hq/lib/python3.8/site-packages/django/db/models/manager.py", line 82, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
  File "/home/skelly/.virtualenvs/hq/lib/python3.8/site-packages/django/db/models/query.py", line 422, in create
    obj.save(force_insert=True, using=self.db)
  File "/home/skelly/.virtualenvs/hq/lib/python3.8/site-packages/django/db/models/base.py", line 743, in save
    self.save_base(using=using, force_insert=force_insert,
  File "/home/skelly/.virtualenvs/hq/lib/python3.8/site-packages/django/db/models/base.py", line 780, in save_base
    updated = self._save_table(
  File "/home/skelly/.virtualenvs/hq/lib/python3.8/site-packages/django/db/models/base.py", line 873, in _save_table
    result = self._do_insert(cls._base_manager, using, fields, update_pk, raw)
  File "/home/skelly/.virtualenvs/hq/lib/python3.8/site-packages/django/db/models/base.py", line 910, in _do_insert
    return manager._insert([self], fields=fields, return_id=update_pk,
  File "/home/skelly/.virtualenvs/hq/lib/python3.8/site-packages/django/db/models/manager.py", line 82, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
  File "/home/skelly/.virtualenvs/hq/lib/python3.8/site-packages/django/db/models/query.py", line 1186, in _insert
    return query.get_compiler(using=using).execute_sql(return_id)
  File "/home/skelly/.virtualenvs/hq/lib/python3.8/site-packages/django/db/models/sql/compiler.py", line 1377, in execute_sql
    cursor.execute(sql, params)
  File "/home/skelly/.virtualenvs/hq/lib/python3.8/site-packages/sentry_sdk/integrations/django/__init__.py", line 489, in execute
    return real_execute(self, sql, params)
  File "/home/skelly/.virtualenvs/hq/lib/python3.8/site-packages/django/db/backends/utils.py", line 67, in execute
    return self._execute_with_wrappers(sql, params, many=False, executor=self._execute)
  File "/home/skelly/.virtualenvs/hq/lib/python3.8/site-packages/django/db/backends/utils.py", line 76, in _execute_with_wrappers
    return executor(sql, params, many, context)
  File "/home/skelly/.virtualenvs/hq/lib/python3.8/site-packages/django/db/backends/utils.py", line 84, in _execute
    return self.cursor.execute(sql, params)
  File "/home/skelly/.virtualenvs/hq/lib/python3.8/site-packages/django/db/utils.py", line 89, in __exit__
    raise dj_exc_value.with_traceback(traceback) from exc_value
  File "/home/skelly/.virtualenvs/hq/lib/python3.8/site-packages/django/db/backends/utils.py", line 84, in _execute
    return self.cursor.execute(sql, params)
django.db.utils.IntegrityError: null value in column "user_id" violates not-null constraint
DETAIL:  Failing row contains (106, 2021-05-31 15:15:50.303867+00, 192, rudolph@qwerty.commcarehq.org, 3, deleted_via: api, 10, null).
```